### PR TITLE
fix #6584 os.getcwd() FileNotFound

### LINKED ISSUE #6584: os.getcwd() FileNotFoundError
```
error: FileNotFoundError(2, 'No such file or directory')
command: /root/anaconda3/bin/conda shell.posix deactivate
user_agent: conda/4.4.4 requests/2.18.4 CPython/3.6.3 Linux/4.13.0-kali1-amd64 kali/2017.3 glibc/2.25
_messageid: -9223372036844975611
_messagetime: 1514233339489 / 2017-12-25 14:22:19

Traceback (most recent call last):
  File "/root/anaconda3/lib/python3.6/site-packages/conda/cli/main.py", line 110, in main
    from ..activate import main as activator_main
  File "/root/anaconda3/lib/python3.6/site-packages/conda/activate.py", line 12, in <module>
    context.__init__()  # oOn import, context does not include SEARCH_PATH. This line fixes that.
  File "/root/anaconda3/lib/python3.6/site-packages/conda/base/context.py", line 214, in __init__
    argparse_args=argparse_args)
  File "/root/anaconda3/lib/python3.6/site-packages/conda/common/configuration.py", line 723, in __init__
    self._set_search_path(search_path)
  File "/root/anaconda3/lib/python3.6/site-packages/conda/common/configuration.py", line 729, in _set_search_path
    self._set_raw_data(load_file_configs(search_path))
  File "/root/anaconda3/lib/python3.6/site-packages/conda/common/configuration.py", line 366, in load_file_configs
    expanded_paths = tuple(expand(path) for path in search_path)
  File "/root/anaconda3/lib/python3.6/site-packages/conda/common/configuration.py", line 366, in <genexpr>
    expanded_paths = tuple(expand(path) for path in search_path)
  File "/root/anaconda3/lib/python3.6/site-packages/conda/common/path.py", line 49, in expand
    return abspath(expanduser(expandvars(path)))
  File "/root/anaconda3/lib/python3.6/posixpath.py", line 374, in abspath
    cwd = os.getcwd()
FileNotFoundError: [Errno 2] No such file or directory
```

### PR DESCRIPTION
**Pull request by kalefranz**
fix #6584

### CODE REVIEW

File: conda/base/context.py [hunk 1]
```diff
@@ -30,6 +31,16 @@
 except ImportError:  # pragma: no cover
     from .._vendor.toolz.itertoolz import concat, concatv, unique
 
+try:
+    os.getcwd()
+except (IOError, OSError) as e:
+    if e.errno == ENOENT:
+        # FileNotFoundError can occur when cwd has been deleted out from underneath the process.
+        # To resolve #6584, let's go with setting cwd to $HOME, and see how far we get.
+        os.chdir(expand('~'))
```
Comment (nehaljwani) [inline conda/base/context.py]: What if the user has no home directory? (Service account or deployment user or something similar)

Should we change it to a directory which is guaranteed to exist? `sys.prefix` maybe?

Comment (kalefranz) [reply conda/base/context.py]: I think that's a great idea.

**Comment by mbargull** [comment]
Exotic use case consideration: If `conda.cli.python_api` is used, this might cause an unexpected `chdir` when importing `conda.context` in a third party application which uses `conda.cli.python_api` (and hence is using the same Python process).

**Comment by nehaljwani** [comment]
But then again, that will happen only if the present directory doesn't exist anymore, as the `os.chdir()` is conditional here. So this seems like a super exotic case to me.

**Comment by mbargull** [comment]
Yep, extremely exotic 😆 -- honestly, I can hardly image we'll ever see anyone experience this.
So maybe only take this as a reminder that with `conda.cli.python_api` in-process global state changes should be done cautiously :wink:

**Pull request merged**